src/components/SelectionList/CheckboxListItem.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import {View} from 'react-native';
+import _ from 'underscore';
+import PressableWithFeedback from '../Pressable/PressableWithFeedback';
+import styles from '../../styles/styles';
+import Text from '../Text';
+import {checkboxListItemPropTypes} from './selectionListPropTypes';
+import Checkbox from '../Checkbox';
+import Avatar from '../Avatar';
+import OfflineWithFeedback from '../OfflineWithFeedback';
+import useLocalize from '../../hooks/useLocalize';
+
+function CheckboxListItem({item, isFocused = false, onSelectRow, onDismissError = () => {}}) {
+    const {translate} = useLocalize();
+    const hasError = !_.isEmpty(item.errors);
+
+    return (
+        <OfflineWithFeedback
+            onClose={() => onDismissError(item)}
+            pendingAction={item.pendingAction}
+            errors={item.errors}
+            errorRowStyles={styles.ph5}
+        >
+            <PressableWithFeedback
+                style={[styles.peopleRow, styles.userSelectNone, isFocused && styles.sidebarLinkActive, hasError && styles.borderColorDanger]}
+                onPress={() => onSelectRow(item)}
+                disabled={item.isDisabled}
+                accessibilityLabel={item.text}
+                accessibilityRole="checkbox"
+                accessibilityState={{checked: item.isSelected}}
+                hoverDimmingValue={1}
+                hoverStyle={styles.hoveredComponentBG}
+                focusStyle={styles.hoveredComponentBG}
+            >
+                <Checkbox
+                    accessibilityLabel={item.text}
+                    disabled={item.isDisabled}
+                    isChecked={item.isSelected}
+                    onPress={() => onSelectRow(item)}
+                    style={item.isDisabled ? styles.buttonOpacityDisabled : {}}
+                />
+                {item.avatar && (
```

Comment:
We should have a better check here. Also Boolean wrap for it.

Reply:
What would be the default value here for the avatar?

Reply:
- Added Boolean wrap
- Made `avatar.source` required in propTypes
- and took avatar props using lodashGet for safer access and default fallback

The default value comes from the pages itself, as they control the `sections` prop, which has the items inside.

In `OptionListUtils.formatMemberForList`, which I'm using in the refactored pages, there is also a default for the avatar: `UserUtils.getDefaultAvatar`

Reply:
Though we currently use only avatar, classification with `iconType` as we do in Menuitem makes more sense than rather making required. Because this is a generic component.

Reply:
@Santhosh-Sellavel I disagree, `CheckboxListItem` is not a generic component in the sense that `MenuItem` is. `<CheckboxListItem />` is called only once inside the SelectionList and is not expected to be called anywhere else

Reply:
@thiagobrez Don't you see the point?

`SelectionList` is the generic component, which had a checklist item as one of its types of list item. Which makes `CheckboxListItem` for all list migrations or new lists.

Reply:
@thiagobrez bump

Reply:
@Santhosh-Sellavel The `avatar` prop is not required, but `avatar.source` is. So, passing an `avatar` is not obligatory, but if you do, then you must guarantee the `source` field inside it exists

```jsx
        avatar: PropTypes.shape({
            source: PropTypes.oneOfType([PropTypes.string, PropTypes.func]).isRequired,  // <-- only this field is required
            name: PropTypes.string,
            type: PropTypes.string,
        }), // <-- the whole avatar is not
```

But if you think this is a blocker I can remove it and leave as it was before - doesn't change much

Reply:
@cristipaval What's your thoughts here?